contrib/init/dogecoind.opt.service

```
@@ -0,0 +1,74 @@
+# This variation of the unit file is for "opt" add-on installations that do not form part of the default installation.
+# (i.e. out of band installations by the user, not installed by a system package manager like "apt" via official sources)
+#
+# The relevant paths are:
+#
+#/opt/dogecoind/dogecoind
+#/etc/opt/dogecoind/
+#/var/opt/dogecoind/
```

I think this should be:
```suggestion
/opt/dogecoin/bin/dogecoind
/opt/dogecoin/etc/
/opt/dogecoin/data/
```

---

Just to clarify, should I change the package directory name from `dogecoind` to `dogecoin` across all the unit files? This probably makes sense since it's not just the daemon that's installed in there.

Also, after re-reading the [LFH /opt/](https://tldp.org/LDP/Linux-Filesystem-Hierarchy/html/opt.html) page, it has this to say:

> "Programs to be invoked by users are located in the directory /opt/'package'/bin. If the package includes UNIX manual pages, they are located in /opt/'package'/man and the same substructure as /usr/share/man must be used. Package files that are variable must be installed in /var/opt. Host-specific configuration files are installed in /etc/opt."

So perhaps the paths should actually be:

- `/opt/dogecoin/bin/dogecoind`
- `/etc/opt/dogecoin/`
- `/var/opt/dogecoin/`

And any manual files would go in `/opt/dogecoin/man/`.

I think this is because historically `/opt/` `/etc/` and `/var/` would often be mounted on different partitions/drives which may have had different characteristics (`/opt/` could technically be mounted read-only).

However if you'd like to go with `/opt/dogecoin/bin/`, `/opt/dogecoin/etc/`, `/opt/dogecoin/data/` because it keeps things closer together, happy to go with that.

There's also this interesting line:

> "Generally, all data required to support a package on a system must be present within /opt/'package', including files intended to be copied into /etc/opt/'package' and /var/opt/'package' as well as reserved directories in /opt. The minor restrictions on distributions using /opt are necessary because conflicts are possible between distribution installed and locally installed software, especially in the case of fixed pathnames found in some binary software. "

Which doesn't have implications for this sytemd file, but does have implications for how the software is installed.

---

> Just to clarify, should I change the package directory name from `dogecoind` to `dogecoin` across all the unit files?

Yes, the binary is called `dogecoind`, the package is called "dogecoin" or "Dogecoin Core" (currently we use "dogecoin" everywhere).

> Which doesn't have implications for this sytemd file, but does have implications for how the software is installed.

This is what I had in mind for the opt:

```
/opt/dogecoin-<version>  # target of dogecoin-<version>-<platform>.tgz
          |-- bin/       # from dogecoin-<version>-<platform>.tgz
          |-- data/      # instead of ~/.dogecoin
          |-- etc/       # from dogecoin-<version>-<platform>.tgz (if we include it)
          |-- include/   # from dogecoin-<version>-<platform>.tgz
          |-- lib/       # from dogecoin-<version>-<platform>.tgz
          |-+ share/     # from dogecoin-<version>-<platform>.tgz
            |-- man/
            |-- systemd/ # ... (if we include it)
/opt/dogecoin            # symlink to /opt/dogecoin-<version>
```

I would normally not do something like `ln -s /opt/dogecoin-<version>/etc /etc/opt/dogecoin`, because /opt/dogecoin would already be symlinked... but maybe that would be the purist thing to do? I don't recall any other software requiring me to do this either fwiw.

Regarding `./data/`, it's hard to predict how people scale `/var`: if you use a great partitioning scheme you would not want logs (in `/var`) to interfere with working data, so if I would be deploying a dbms, I'd mount the data files (and depending on the flavor, redo logs) on their own partition, outside of `/var`, eg `/data`... However, I suspect most people are running their dogecoind on a VPS without any partitioning at all, so it doesn't matter where we put it. If someone would want to put their data on a separate partition, perhaps we should just write an example in documentation for that?

---

One last thing before I change the /opt/ directories, if the configuration and data directory is stored within `/opt/dogecoin-<version>`, how are upgrades handled? Does the installer copy/move the old configuration and data directory out of the old install, or is it up to the user to do this?

---

I've thought more about this and the only issue with keeping the data directory under the software directory - since we intend to symlink it, is indeed portability of that data cross-version. So perhaps `/var/opt/dogecoin` IS better than `/opt/dogecoin-<version>/data`, we just need to document that `/var/` then needs to have 55Gb+ capacity for storing the chain for a fully validating node, or approximately 4Gb for a pruning node.

---

`/etc/opt/dogecoin` also avoids the same issue for the daemon configuration during upgrades. The `/opt/dogecoin` directory should still contain a copy of the default configuration for the current version, however. With that, dogecoin would follow the FHS pretty faithfully.

> we just need to document that /var/ then needs to have 55Gb+ capacity for storing the chain for a fully validating node, or approximately 4Gb for a pruning node.

Absolutely - 55GB is pretty chonky. Then the user can mount `/var/` or `/var/opt/dogecoin` however they like, just like `/var/log`, /var/tmp`, etc.